tests/test_packages/test_connections/test_p2p_libp2p/test_fault_tolerance.py

```diff
@@ -19,226 +19,157 @@
+        assert not self.multiplexer_relay.is_connected
+
+        with pytest.raises(Empty):
+            attempt_sending()
 
-        # currently, multiplexer cannot be restarted
-        self.multiplexer_relay = Multiplexer([self.relay], protocols=[DefaultMessage])
         self.multiplexer_relay.connect()
         self.change_state_and_wait(self.multiplexer_relay, expected_is_connected=True)
-        self.multiplexers.append(self.multiplexer_relay)
-
-        envelope = self.enveloped_default_message(to=to, sender=sender)
-        time.sleep(10)
-        self.multiplexer_genesis.put(envelope)
-
-        delivered_envelope = self.multiplexer.get(block=True, timeout=TIMEOUT)
-
-        assert delivered_envelope is not None
-        assert self.sent_is_delivered_envelope(envelope, delivered_envelope)
+        assert self.multiplexer_relay.is_connected
+        attempt_sending()
 
     def test_envelope_routed_from_client_after_relay_restart(self):
         """Test envelope routed from third relay client after relay restart."""
 
-        to = self.connection.address
+        to = self.connection1.address
         sender = self.connection2.address
 
-        envelope = self.enveloped_default_message(to=to, sender=sender)
-        self.multiplexer2.put(envelope)
-        delivered_envelope = self.multiplexer.get(block=True, timeout=TIMEOUT)
+        def attempt_sending():
+            envelope = self.enveloped_default_message(to=to, sender=sender)
+            self.multiplexer2.put(envelope)
+            delivered_envelope = self.multiplexer1.get(block=True, timeout=TIMEOUT)
+            assert delivered_envelope is not None
+            assert self.sent_is_delivered_envelope(envelope, delivered_envelope)
 
-        assert delivered_envelope is not None
-        assert self.sent_is_delivered_envelope(envelope, delivered_envelope)
+        assert self.multiplexer_relay.is_connected
+        attempt_sending()
 
         self.multiplexer_relay.disconnect()
         self.change_state_and_wait(self.multiplexer_relay, expected_is_connected=False)
+        assert not self.multiplexer_relay.is_connected
+
+        with pytest.raises(Empty):
+            attempt_sending()
 
-        # currently, multiplexer cannot be restarted
-        self.multiplexer_relay = Multiplexer([self.relay], protocols=[DefaultMessage])
         self.multiplexer_relay.connect()
         self.change_state_and_wait(self.multiplexer_relay, expected_is_connected=True)
-        self.multiplexers.append(self.multiplexer_relay)
-
-        envelope = self.enveloped_default_message(to=to, sender=sender)
-        time.sleep(10)
-        self.multiplexer2.put(envelope)
-        delivered_envelope = self.multiplexer.get(block=True, timeout=TIMEOUT)
-
-        assert delivered_envelope is not None
-        assert self.sent_is_delivered_envelope(envelope, delivered_envelope)
-
-
-@libp2p_log_on_failure_all
-class TestLibp2pConnectionRelayNodeRestartOutgoingEnvelopes(BaseTestLibp2pRelay):
```

Comment:
see comment above, merged test class for incoming and outgoing messages after restart